eng/common/scripts/GeneralReleaseNotesLogic.ps1

```diff
@@ -0,0 +1,128 @@
+            continue
+        }
+        if ($line -eq "-->")
+        {
+            break
+        }
+        if ($checkLine)
+        {
+            $pkgInfo = ($line.Trim()).Split(":")
+            $packageName = $pkgInfo[0]
+            $packageMetaData = $CsvMetaData | Where-Object { $_.Package -eq $packageName }
+            if ($packageMetaData.Count -gt 0)
+            {
+                $presentPkgInfo += $line.Trim()
+            }
+        }
+    }
+    return $presentPkgInfo
+}
+
+function Filter-ReleaseHighlights ($releaseHighlights)
+{
+    $results = @{}
+
+    foreach ($key in $releaseHighlights.Keys)
+    {
+        $keyInfo = $key.Split(":")
+        $packageName = $keyInfo[0]
+        $packageVersion = $keyInfo[1]
+
+        $packageMetaData = $CsvMetaData | Where-Object { $_.Package -eq $packageName }
+
+        if ($packageMetaData.ServiceName -eq "template")
+        {
+            continue
+        }
+
+        $existingPackages = GetExistingPackageVersions -PackageName $packageName `
+        -GroupId $packageMetaData.GroupId
+
+        $versionExists = $existingPackages | Where-Object { $_ -eq $packageVersion }
+
+        if ($null -eq $versionExists)
+        {
+            continue
+        }
+
+        $results.Add($key, $releaseHighlights[$key])
+    }
+    return $results
+}
+
+function Write-GeneralReleaseNote ($releaseHighlights, $releaseFilePath)
+{
+    $releaseContent = Get-Content $releaseFilePath
+    $newReleaseContent = @()
+    $writingPaused = $False
+
+    foreach ($line in $releaseContent)
+    {
+        if ($line -match $PATTERN_REGEX)
+        {
+            $sectionName = $matches["SectionName"]
+            $pattern = $matches["Pattern"]
+
+            foreach ($key in $releaseHighlights.Keys)
+            {
+                $pkgInfo = $key.Split(":")
+                $packageName = $pkgInfo[0]
+                $packageVersion = $pkgInfo[1]
+                $packageFriendlyName = ($csvMetaData | Where-Object { $_.Package -eq $PackageName }).DisplayName
+
+                if ($null -eq $packageFriendlyName)
+                {
+                    $packageFriendlyName = $packageName
+                }
+
+                $changelogUrl = $releaseHighlights[$key]["ChangelogUrl"]
+                $changelogUrl = "(${changelogUrl})"
+                $highlightsBody = $releaseHighlights[$key]["Content"]
+                $packageSemVer = [AzureEngSemanticVersion]::ParseVersionString($PackageVersion)
+                
+                $lineValue = $ExecutionContext.InvokeCommand.ExpandString($pattern)
+                if ([System.String]::IsNullOrEmpty($sectionName))
+                {
+                    $newReleaseContent += $lineValue
+                }
+                elseif ($packageSemVer.VersionType -eq $sectionName)
+                {
+                    $newReleaseContent += $lineValue
+                }
+            }
+            $newReleaseContent += ""
+            if ($writingPaused)
```

Comment:
I'm not sure I understand the writing paused logic. Can you help me understand what that is for?

Reply:
I did this to properly accommodate the `bash` section. Basically I don't want to write the closing "```" till am done writing the install instruction information.